src/mmw/js/src/modeling/models.js

```diff
@@ -464,8 +464,11 @@ var ProjectModel = Backbone.Model.extend({
                 mapshed_data: mapshedData,
             }, payload))
         }).done(function(result) {
-            self.set('hydroshare', result);
-            self.set('hydroshare_errors', []);
+            self.set({
+                hydroshare: result,
+                hydroshare_errors: [],
+                is_private: false,
```

Comment:
This is a little unconventional in its assumption that the server is also updating this field, but I understand that receiving the updated project from the server to kick off this model change might require some refactoring that it's maybe not worth investing in now.

Reply:
Agreed. While a refactoring would be expensive, I will add a comment for clarity.